digitalfabrik/entitlementcard
# 1419: Create card form

### Short description
As a user i want to create and activate my koblenz pass



### Proposed changes
- add extension reference number
- improve a11y for extensions in mobile mode for koblenz (add clear button, improve size)
- add 3 step form to create, inform and activate a pass
- add a new hook to create cards from self service and create pdf separately uses different endpoint and states.
- make region optional in pdfs
- extend CardBlueprint by SelfServiceCard to make initialization with region optional

### Side effects



- hopefully none, please test properly also card and pdf creation for nuernberg and bavaria (csv and single)

### TODOS
- decided to create tests in separate issue, since its getting too big #1646 
- DeepLinks are not correct since buildConfig for app is missing #1647 
- Store links are missing since they don't exists and may be added to app buildConfig #1647
- There may be also added a help section in the header but its not decided yet
- Information about reference number info dialog #1649
- Adjust self service texts #1650

### Testdata

```
regionKey,userHash,startDate,endDate,revoked
07111,"$argon2id$v=19$m=19456,t=2,p=1$cr3lP9IMUKNz4BLfPGlAOHq1z98G5/2tTbhDIko35tY",01.01.2024,01.01.2025,false
07111,"$argon2id$v=19$m=19456,t=2,p=1$POG+rDhWlxSW7ItkP9hZRHqr46u6a6TQO1Ib2X+khMA",01.01.2024,01.01.2025,false
07111,"$argon2id$v=19$m=19456,t=2,p=1$zf7vqckgmYBPbjjW1CwqudF+OjHmo2EifgxhuuVHFs0",01.01.2024,01.01.2025,true
```

### Testdata with base information for testing
```
regionKey,userHash,startDate,endDate,revoked,birthday,referenceNumber
07111,"$argon2id$v=19$m=19456,t=2,p=1$cr3lP9IMUKNz4BLfPGlAOHq1z98G5/2tTbhDIko35tY",01.01.2024,01.01.2025,false, 10.06.2003(12213), 123K
07111,"$argon2id$v=19$m=19456,t=2,p=1$POG+rDhWlxSW7ItkP9hZRHqr46u6a6TQO1Ib2X+khMA",01.01.2024,01.01.2025,false, 10.06.2003(12213), 5.021.025.688
07111,"$argon2id$v=19$m=19456,t=2,p=1$zf7vqckgmYBPbjjW1CwqudF+OjHmo2EifgxhuuVHFs0",01.01.2024,01.01.2025,true, 28.05.2003(12200), 000D000001
```

### Testing (mobile mode)
1. Import users via curl or postman
2. Ensure in the backend config `selfServiceEnabled: true` is set.
`curl -F file=@import_testdata_local.csv http://0.0.0.0:8000/users/import`
3. go to http://localhost:3000/ and select Koblenz
4. Go to http://localhost:3000/erstellen
5. Use the infos from the "Testdata with base information" and type in the data in the form. Note that the name will not be hashed, so you can choose whatever you want.
6. Check if the pass can be created and check error message if the infos in the form were not matching
7. Store Links are not defined yet so klick "Zur Aktivierung"
8. Try to download the pdf and check the details
9. Klick on activate card will not work since app buildConfig is missing.

### Resolved issues



Fixes: #1419
Fixes: #1420
